kraft-jennifer-ndvi-automation.py

```diff
@@ -0,0 +1,698 @@
+    vals: list
+        A list of values needed to create the cloud mask
+
+
+    Returns
+    -----------
+    ndvi_crop : Xarray Dataset
+        a cropped and masked xarray object containing NDVI values
+    """
+
+    crop_json = crop_bound.geometry
+
+    # Clip pixel qa cloud mask layer
+    cl_mask_crop = pixel_qa.rio.clip(crop_json)
+
+    # Calculate NDVI
+    ndvi_xr = (all_bands[1]-all_bands[0]) / (all_bands[1]+all_bands[0])
+    # Clip NDVI layer
+    ndvi_crop = ndvi_xr.rio.clip(crop_json)
+    
+    # Apply cloud mask to NDVI
+    ndvi_crop = ndvi_crop.where(~cl_mask_crop.isin(vals))
+
+    return ndvi_crop
+
+
+# In[6]:
+
+
+# Get a list of each directory
+path = os.path.join("ndvi-automation", "sites")
+
+# Get a list of both site directories (We will talk more about automation next week)
+sites = glob(path + "/*/")
+sites
+
+# Get the site name
+site_name = os.path.basename(os.path.normpath(sites[0]))
+site_name
+
+# Open up the shapefile for clipping your landsat data to the study area
+vector_dir = os.path.join(sites[0],
+                          "vector")
+
+# Open crop boundary
+site_boundary_path = os.path.join(vector_dir,  site_name + "-crop.shp")
+crop_bound = gpd.read_file(site_boundary_path)
+crop_bound.plot()
+plt.show()
+
+landsat_dir = os.path.join(sites[0],
+                           "landsat-crop")
+# This is the crop folder containing all of the .tif files
+landsat_dir
+
+#Create lists of subdirectories
+landsat_dirs = sorted(glob(os.path.join(landsat_dir, "LC08*")))
+landsat_dirs
+
+# Select just a single directory and grab bands 4-5 from the directory
+adir = landsat_dirs[4]
+
+# Open bands
+band_paths = sorted(glob(os.path.join(adir, "*band*[4-5].tif")))
+band_paths
+
+# Open & clip the data - a single band using rioxarray
+band = rxr.open_rasterio(band_paths[0], masked=True).rio.clip(crop_bound.geometry,
+                                                              from_disk=True).squeeze()
+
+# Specify the valid range of values for landsat
+valid_range = (0, 10000)
+
+if valid_range:
+    mask = ((band < valid_range[0]) | (band > valid_range[1]))
+    band = band.where(~xr.where(mask, True, False))
+
+band
+
+# Open and clean a single band using a function
+band_4 = open_clean_bands(band_path=band_paths[0],
+                          crop_extent=crop_bound,
+                          # The range of valid values for landsat can include negative values
+                          # for this week let's stick with 0-10000
+                          valid_range=(0, 10000))
+
+# %%timeit
+
+# Open and clean a single band
+band_4 = open_clean_bands(band_path=band_paths[0],
+                          crop_extent=crop_bound,
+                          # The range of valid values for landsat can include negative values
+                          # for this week let's stick with 0-10000
+                          valid_range=(0, 10000))
```

Comment:
These functions are duplicated

Reply:
Oops! If I recall correctly, this entire cell was a practice/workflow cell, working on processing individual bands before applying functions for the actual assignment itself. I should have either removed this cell, or clearly labeled it as a workspace. And regardless, I could have cleaned up this duplication. Thank you!